FLEDGE_implementation_overview.md

```diff
@@ -0,0 +1,66 @@
+# Protected Audience Implementation Overview
+This document is intended to give a rough overview of the Protected Audience implementation in Chrome, with a focus towards performance-relevant design decisions, and timeouts.  The purpose is to both explain how these work, and help consumers figure out if different timeouts/timeout behavior would be useful.
+
+This document starts out discussing singleton auctions, without any nested component auctions.  There is a section on component auctions at the end.
+
+
+## Phase 1 Load Interest Groups
+Immediately after runAdAuction() is invoked, all interest groups, k-anon data, and win history will be loaded from the SQLite database.  No worklets will be loaded until after this is complete, for all interestGroupBuyers participating in an auction.  Interest groups are loaded on a per-buyer basis, and remain grouped by buyer for the entire auction.  Interest groups that cannot participate in the auction (due to, e.g., no biddingLogicURL) are removed.  If there are no interest groups that can bid in the auction, and no additional bids, the auction completes without a winner, bypassing future stages.
+
+For each bidder origin, Chrome determines a tentative priority - using each interest group’s fixed priority, or priority vector, if present.  Chrome then sorts the interest groups based on priority, removing any with priority vectors and negative priorities.  Interest groups with matching priorities are grouped by joining origin (to improve performance in case ExecutionMode is groupByOrigin).  Groups with matching joining origins and priority are randomly shuffled.  For each bidder origin, Chrome checks if any interest group participating in the auction has “enableBiddingSignalsPrioritization” set.  If not, Chrome applies the auction’s per-buyer size limit to the bidder’s interest groups.  Either way, the interest groups are now in a priority order that will affect the order in which other resources will be fetched, and worklets will be run in.  None of this affects additional bids.
```

Comment:
>each bidder origin

Suggest s/bidder/buyer for consistency with PA terminology.

Reply:
Did a little replacing here.  I guess in the explainer, at least, we generally use "buyer" for AuctionConfig fields, which affect all IGs owned by the same "buyer", while we often use "bidder" when referring to interest groups.  This document is probably still pretty inconsistent, and some cases it's ambiguous which the text should be referring to.